src/source/kafka.rs

```diff
@@ -243,8 +243,14 @@ impl Source for Int {
                         m.offset().to_string(),
                     ];
                     let data = data.to_vec();
+                    let mut kafka_meta_data = Value::object_with_capacity(1);
+                    kafka_meta_data.insert("kafka", Value::object_with_capacity(2))?;
```

Comment:
keep this value around and only insert it when you handled key and headers, otherwise you need to do unneccessary lookups.